src/dist/component/package.rs

```diff
@@ -163,19 +166,40 @@ struct MemoryBudget {
     used: usize,
 }
 
+lazy_static! {
+    static ref RAM_NOTICE_SHOWN: AtomicBool = AtomicBool::new(false);
+}
+
```

Comment:
I'm not *certain* that this needs `lazy_static!` -- I think AtomicBool is meant to be used as a synchronisation primitive.  Certainly a lot of its methods take `&self` rather than `&mut self`

Reply:
ah, I see, right, iterating.